Switch.js

```diff
@@ -58,60 +58,67 @@ define([
 		},
 
 		_pointerDownHandler: function (e) {
-			this._startX = this._curX = e.clientX;
-			pointer.setPointerCapture(this._knobGlassNode, e.pointerId);
-			if (!this._pHandlers) {
-				this._pHandlers = [
-					{e: "pointermove", l: this._pointerMoveHandler.bind(this)},
-					{e: "pointerup", l: this._pointerUpHandler.bind(this)},
-					{e: "lostpointercapture", l: this._lostPointerCaptureHandler.bind(this)}
-				];
+			if (! this.disabled) {
+				this._startX = this._curX = e.clientX;
+				pointer.setPointerCapture(this._knobGlassNode, e.pointerId);
+				if (!this._pHandlers) {
+					this._pHandlers = [
+						{e: "pointermove", l: this._pointerMoveHandler.bind(this)},
+						{e: "pointerup", l: this._pointerUpHandler.bind(this)},
+						{e: "lostpointercapture", l: this._lostPointerCaptureHandler.bind(this)}
+					];
+				}
+				this._pHandlers.forEach(function (h) { this._knobGlassNode.addEventListener(h.e, h.l); }.bind(this));
+				e.preventDefault();
+				e.stopPropagation();
 			}
-			this._pHandlers.forEach(function (h) { this._knobGlassNode.addEventListener(h.e, h.l); }.bind(this));
-			e.preventDefault();
-			e.stopPropagation();
 		},
 
 		_pointerMoveHandler: function (e) {
-			var dx = e.clientX - this._curX,
-				cs = window.getComputedStyle(this._pushNode),
-				w = parseInt(cs.width, 10);
-			if (!this._drag && Math.abs(e.clientX - this._startX) > 4) {
-				this._drag = true;
-				$(this._innerNode).removeClass("-d-switch-transition");
-				$(this._pushNode).removeClass("-d-switch-transition");
-				$(this._innerWrapperNode).removeClass("-d-switch-transition");
+			if (! this.disabled) {
```

Comment:
this is not strictly enforced but I think our policy is to not put spaces between `!` operator and the expression? (same at another place below).